### PR TITLE
Add java bindings for distinct count

### PR DESCRIPTION
**Pull request by revans2**
## Description
This adds in java bindings for cudf::distinct_count for both table and column APIs.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

**Review by gerashegalov — APPROVED** [review]
LGTM

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -3999,6 +4015,8 @@ static DeviceMemoryBufferView getOffsetsBuffer(long viewHandle) {
   }
 
   // Native Methods
+  private static native int distinctCount(long handle, boolean nullsIncluded, boolean nansAreNulls);
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:4018]: Why do we need this parameter? I don't see any public Java API calling it. If we are not using it then should we just remove it completely from Java code?

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 2]
```diff
@@ -3967,6 +3967,22 @@ static long getValidityBufferSize(int numRows) {
...
+   * Count how many rows in the column are distinct from one another.
+   * @param nullPolicy if nulls should be included or not.
+   */
+  public int distinctCount(NullPolicy nullPolicy) {
+    return distinctCount(getNativeView(), nullPolicy.includeNulls, false);
+  }
+
+  /**
+   * Count how many rows in the column are distinct from one another.
+   * Nulls are included.
```
[the remark this answers is not in the document]
Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:3980]: What parameter?  The false in the native code?

File: java/src/main/java/ai/rapids/cudf/ColumnView.java (hunk 1 above)
Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:4018]: I added it thinking that there was a java enum that corresponded to it, but never got around to adding in the enum. I can remove it if we want to.

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:4018]: Yes please remove it.

**Review by ttnghia — APPROVED** [review]

**Pull request merged**